include/xgboost/tree_updater.h

```diff
@@ -91,22 +90,23 @@ class TreeUpdater : public Configurable {
     return false;
   }
 
-  virtual char const* Name() const = 0;
+  [[nodiscard]] virtual char const* Name() const = 0;
 
-  /*!
+  /**
    * \brief Create a tree updater given name
    * \param name Name of the tree updater.
    * \param ctx A global runtime parameter
+   * \param task Infomation about the objective.
    */
-  static TreeUpdater* Create(const std::string& name, Context const* ctx, ObjInfo task);
+  static TreeUpdater* Create(const std::string& name, Context const* ctx, ObjInfo const* task);
```

Comment:
```suggestion
  static TreeUpdater* Create(const std::string& name, Context const* ctx, ObjInfo const& task);
```

Have you considered using a constant reference instead? I don't see a reason for allowing a `nullptr` here.

Reply:
using a pointer can ask the compiler to check whether I'm accidentally passing it by value. If we change it to reference, we need to disable the default copy constructor and assignment operator, then define an explicit copy method to have the same effect, which seems to be a little too much for a simple struct.

Reply:
Got it, then pointer is fine.